flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -1035,19 +868,37 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
      * @param message {{text, isAmber, isTypeTwo}} Message Object
      * @param isResolved {function} Function that determines if the error is resolved at this moment (type II only).
      * @param onClear {function} Function that executes when the error is actively cleared by the pilot (type II only).
+     * @param propagate {boolean} Whether to propagate type II messages to the other MCDU
      */
     addNewMessage(message, isResolved = () => {
         return false;
-    }, onClear = () => {}) {
+    }, onClear = () => {}, propagate = true) {
         if (message.isTypeTwo) {
             if (!isResolved()) {
                 this._addTypeTwoMessage(message.text, message.isAmber, isResolved, onClear);
+                if (propagate) {
+                    this.sendMessageOffside(message, isResolved, onClear);
+                }
             }
         } else {
             this._showTypeOneMessage(message.text, message.isAmber);
         }
     }
 
+    /**
+     * Send a scratchpad message to the offside MCDU
+     * @param message {{text, isAmber, isTypeTwo}} Message Object
+     * @param isResolved {function} Function that determines if the error is resolved at this moment (type II only).
+     * @param onClear {function} Function that executes when the error is actively cleared by the pilot (type II only).
+     */
+    sendMessageOffside(message, isResolved, onClear) {
+        if (this.side === "L") {
+            this.fmc.addNewMessageRight(message, isResolved, onClear);
+        } else {
+            this.fmc.addNewMessageLeft(message, isResolved, onClear);
+        }
+    }
```

Comment:
**thought**: It might be good software design to make the MCDU unaware of which side/instance it is.
**suggestion**: Here and in other locations, move the responsibility of sending data to the other side(s) up.
```javascript
this.fmc.addNewMessageToOthers(this, message, isResolved, onClear);
```